control-plane/plugin/src/resources/node.rs

```diff
@@ -239,6 +336,126 @@ impl GetHeaderRow for NodeDisplay {
+                        Ok(node) => {
+                            let node_body = &node.clone().into_body();
+
+                            match &node_body.spec.as_ref().unwrap().cordondrainstate {
+                                Some(ds) => match ds {
+                                    CordonDrainState::cordonedstate(_) => {
+                                        match output {
+                                            OutputFormat::None => {
+                                                println!("Drain has been cancelled");
+                                            }
+                                            _ => {
+                                                // json or yaml
+                                                print_table(output, node.into_body());
+                                            }
+                                        }
+                                        break;
+                                    }
+                                    CordonDrainState::drainingstate(_) => {}
+                                    CordonDrainState::drainedstate(_) => {
+                                        match output {
+                                            OutputFormat::None => {
+                                                println!("Drain has completed");
+                                            }
+                                            _ => {
+                                                // json or yaml
+                                                print_table(output, node.into_body());
+                                            }
+                                        }
+                                        break;
+                                    }
+                                },
+                                None => {
+                                    match output {
+                                        OutputFormat::None => {
+                                            println!("Drain has been cancelled");
+                                        }
+                                        _ => {
+                                            // json or yaml
+                                            print_table(output, node.into_body());
+                                        }
+                                    }
+                                    break;
+                                }
+                            }
+                        }
+                        Err(e) => {
+                            println!("Failed to get node {}. Error {}", id, e);
+                            break;
+                        }
+                    }
+                    if timeout_instant.is_some() && time::Instant::now() > timeout_instant.unwrap()
+                    {
+                        println!("Drain command timed out");
+                        break;
+                    }
```

Comment:
Question: Is this the time until which the user wants to look at the drain command's output and when time is up we just say the time is up and come out of the loop, but the drain may not be complete by then? What's the significance of this?

Reply:
That's correct. AIUI we want the drain command to block until the drain is completed, but have some option of timing-out the command if drain takes too long.

Reply:
Okay. That makes sense, but I wonder from a user perspective whether a we would set a timeout or rather do a `Ctrl+c` to terminate the command.